antea/scripts/process_mc_petit_FBK.py

```diff
@@ -0,0 +1,307 @@
+
+            if len(evt_tof) == 0:
+                continue
+
+            ### Divide sensors
+            df_sns0 = evt_sns[evt_sns.sensor_id < 100]
+            df_sns2 = evt_sns[evt_sns.sensor_id > 100]
+
+            sns0 = df_sns0.sensor_id.values
+            sns2 = df_sns2.sensor_id.values
+
+            ### Use absolute times in units of ps
+            times = evt_tof.time / units.ps
+              
+            ### Add SiPM jitter, if different from zero
+            if sigma_sipm > 0:
+                times = np.round(np.random.normal(times, sigma_sipm))
+            evt_tof = evt_tof.drop('time', axis=1)
+            evt_tof.insert(len(evt_tof.columns), 'time', np.round(times).astype(int)) # here we have bins of 1 ps
+            evt_tof.insert(len(evt_tof.columns), 'charge', np.ones(len(times)).astype(int))
+
+            ### Produce a TOF dataframe with convolved time response
+            tof_sns = evt_tof.sensor_id.unique()
+
+            evt_tof_exp_dist = []
+            for s_id in tof_sns:
+                tdc_conv    = shf.sipm_shaping_convolution(evt_tof, spe_resp, s_id, time_window)
+                tdc_conv_df = shf.build_convoluted_df(evt, s_id, tdc_conv)
+
+                if sigma_elec > 0:
+                    tdc_conv_df = tdc_conv_df.assign(time=np.random.normal(tdc_conv_df.time.values, sigma_elec))
+
+                tdc_conv_df = tdc_conv_df[tdc_conv_df.charge > timestamp_thr/norm]
+                tdc_conv_df = tdc_conv_df[tdc_conv_df.time == tdc_conv_df.time.min()]
+                evt_tof_exp_dist.append(tdc_conv_df)
+            evt_tof_exp_dist = pd.concat(evt_tof_exp_dist)
+
+
+            try:
+                min_id0, min_t0 = rf.find_first_time_of_sensors(evt_tof_exp_dist, sns0, n_pe)
+            except WaveformEmptyTable:
+                print(f'TOF dataframe has no minimum time in plane 0 for event {evt}')
+                min_id0, min_t0 = [-1], -1
+            try:
+                min_id2, min_t2 = rf.find_first_time_of_sensors(evt_tof_exp_dist, sns2, n_pe)
+            except:
+                print(f'TOF dataframe has no minimum time in plane 2 for event {evt}')
+                min_id2, min_t2 = [-1], -1
+
+            ## select SiPM with max charge
+            max_ch0 = df_sns0.loc[df_sns0.charge.idxmax()]
+            max_ch2 = df_sns2.loc[df_sns2.charge.idxmax()]
+            sid_0   = max_ch0.sensor_id
+            sid_2   = max_ch2.sensor_id
+            ch_0    = max_ch0.charge
+            ch_2    = max_ch2.charge
+            or_0    = max_ch0.original_pe
+            or_2    = max_ch2.original_pe
+
+            t_ch0 = evt_tof_exp_dist[evt_tof_exp_dist.sensor_id == sid_0].time.min()
+            t_ch2 = evt_tof_exp_dist[evt_tof_exp_dist.sensor_id == sid_2].time.min()
+            tmin_ch0.append(t_ch0)
+            tmin_ch2.append(t_ch2)
+            max_charge0.append(ch_0)
+            max_charge2.append(ch_2)
+            max_or0.append(or_0)
```

Comment:
If you save the original charge, shouldn't you apply the fluctuation and the threshold also to it?